compiler/rustc_mir/src/transform/promote_consts.rs

```diff
@@ -758,7 +758,7 @@ impl<'tcx> Validator<'_, 'tcx> {
             ty::FnDef(def_id, _) => {
                 is_const_fn(self.tcx, def_id)
                     || is_unstable_const_fn(self.tcx, def_id).is_some()
-                    || is_lang_panic_fn(self.tcx, self.def_id.to_def_id())
+                    || is_lang_panic_fn(self.tcx, def_id)
```

Comment:
This line was incorrect. It should be checking the callee, not the caller. It should just be removed entirely, since panicking in a const-context is still unstable.